### PR TITLE
Major improvements to the Words functionality

### PR DESCRIPTION
**Pull request by mwhelan**
All of the file dictionaries have been added to the Any API via equivalence class extension methods.

    Any.AddressAusCity();

In *addition*, can now access any built-in or user-defined dictionary with a single interface

	Any.Words(FromDictionary.InternetUrl)
	// E.g. Any.Words(FromDictionary.InternetUrl).Data
	// E.g. Any.Words(FromDictionary.InternetUrl).Next()
	Any.Words("MyCustomDictionary").Data;

This PR **does propose** breaking changes to the API. With the explosion of Any methods, names such as Any.EmailAddress have changed to Any.PersonEmailAddress so that all categories of methods are grouped together with a one-to-one mapping to the name of the file the data originated from. These are subject to agreement by @robdmoore of course.

### CODE REVIEW

**Comment by mwhelan** [comment]
I think the CI server isn't handling a C#6 feature I used on line 28 of the Words class. Sorry, I did try to resist those, but sometimes they just slip in... ;-)

**Comment by robdmoore** [comment]
Drop the C#6 stuff. Happy to look into adding that (and getting CI server ready for it), but definitely would want that in a separate PR.

I'd also prefer to avoid breaking changes for this one. I'm happy with the new grouped names, but let's keep the existing ones too and simply call the new ones.

For things like Any.FirstName that are so obvious / common I like them to be less verbose and more obvious.

We'll also need to get all of these added to the [documentation](http://dossier.teststack.net/docs/anonymous-values-and-equivalence-classes) when merging.

File: TestStack.Dossier.Tests/DataSources/Dictionaries/WordsCacheTests.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.IO;
+using Shouldly;
+using TestStack.Dossier.DataSources.Dictionaries;
+using Xunit;
+
+namespace TestStack.Dossier.Tests.DataSources.Dictionaries
+{
+    public class WordsCacheTests : IDisposable
```
Comment (robdmoore) [inline TestStack.Dossier.Tests/DataSources/Dictionaries/WordsCacheTests.cs]: Any public classes should have xmldoc comments

**Comment by mwhelan** [comment]
Agreed about the C#6. Better to do it in one go, but a few slipped in. ;-) 

So, does that mean keep Any.FirstName and Any.PersonFirstName, where the result is the same?

Do the new names not already make this a breaking change? Or are breaking changes only classed as things that change rather than additional things too?

My only concern with keeping things like Any.FirstName is just how big the interface is. I was even wondering if we could come up with Any.Person.FirstName type things, but some categorisation seems like a good thing

**Comment by robdmoore** [comment]
It's not a breaking change if we add something new to the surface area. Removing something that was there is though.

I'm comfortable with Any.Person.* etc. as well. that might be better for intellisense.

I still like having the existing ones we had with their previous names too though since I think they are common ones that people will want to use so should be quick and easy to get to.

**Comment by mwhelan** [comment]
Awesome. I'm happy with all your suggestions and will try to address them all. Thanks for the great feedback.

**Pull request merged**